### ISSUE TITLE
Problems with named unicode in combination with Fstrings

### ISSUE THREAD

**Issue by xmatthias** [microsoft/pylance-release#161]
## Environment data

-   Language Server version: 2020.7.3
-   OS and version: linux
-   Python version (& distribution if applicable, e.g. Anaconda): python 3.8.2

## Expected behaviour

No error, since the syntax is valid and prints the Greek SUM letter correctly
`(200 Σ)`.

Not having this in an fstring results in no error - so i guess the fstring `{` detection ignores the leading `\N`.

## Actual behaviour

Pylance shows error "Unexpected token at end of expression" for fstring

![2020-07-23-064716_915x230_scrot](https://user-images.githubusercontent.com/5024695/88252963-5cb68f80-ccb0-11ea-9ddb-98bcbefd7afb.png)

## Code Snippet / Additional information

```python

hello = 200
# The following line shows both GREEK and CAPITAL as error
print(f"({hello} \N{GREEK CAPITAL LETTER SIGMA})")

print("\N{GREEK CAPITAL LETTER SIGMA})")

```

**Comment by erictraut** [comment]
Thanks for the bug report. There was a bug in unicode name parsing related to capital letters. A fix will be in the next version of Pylance. While you're waiting for the fix, you can work around the issue by using lower-case characters ("greek capital letter sigma").

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.40, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202074-29-july-2020

**Comment by xmatthias** [comment]
Great, thanks! :+1:
